internal/controller/controlplane/helper.go

```diff
@@ -344,6 +345,28 @@ func (c *K0sController) createAutopilotPlan(ctx context.Context, kcp *cpv1beta1.
 		return nil
 	}
 
+	var existingPlan unstructured.Unstructured
+	err := clientset.RESTClient().Get().AbsPath("/apis/autopilot.k0sproject.io/v1beta2/plans/autopilot").Do(ctx).Into(&existingPlan)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("error getting autopilot plan: %w", err)
+	}
+
+	state, found, err := unstructured.NestedString(existingPlan.Object, "status", "state")
+	if err != nil {
+		return fmt.Errorf("error getting autopilot plan's state: %w", err)
+	}
+	if found {
+		if state == "Schedulable" || state == "SchedulableWait" {
+			// autopilot is already running
+			return nil
+		}
+	}
+
+	err = clientset.RESTClient().Delete().AbsPath("/apis/autopilot.k0sproject.io/v1beta2/plans/autopilot").Do(ctx).Error()
```

Comment:
do we need to delete autopilot plan if we are using PATCH?

Reply:
Unfortunately, yes. Autopilot behaves badly if we update the existing plan. So I got rid of patching and returned the previous POST implementation